protocol.md

```diff
@@ -133,6 +136,12 @@ This protocol is repeated for each recipient.
 Note that the sender has confirmation that the Shard is written to the recipients disk _before_ they delete the outbox copy.
 Simiarly, the recipient receives confirmation that the sender has deleted their copy (by waiting for the wormhole to close).
 
+> We still need the recovery process. Tricky bits: 
+> - How does the custodian know which shard belongs to which secret owner? Public key?
```

Comment:
With PKI I would say "yes, public-key". Without that, maybe "random ID determined by the sender"...?

That is, you send a file named like `shard-<hex encoded random id>.secret` or similar.

However, the current wormhole file-transfer protocol doesn't have a way to "request" a certain file (just to send one).

So, perhaps a "petname" approach would be best: the Custodian decides on a petname of their choosing, and the software keeps the Shard under that name (e.g. you might store a Shard for me under "meejah").

This depends on human memory to map the right petname to the right human -- but perhaps that's actually a good thing and mirrors the "trust" we put in the humans exchanging the codes..?
(The main advantage here of a petname over something automatic is that the existing file-transfer protocol can be leveraged).

One way to cover the "what about many Shards?" case could be to send a file named with a "purpose". That is, I could send you a file named like `"scuttlebutt-recovery.secret"` or similar, and you store it in `.../meejah/scuttlebutt-recovery.secret` .. so when you select "meejah" as a petname to send a Shard back to, the software can ask you to select amongst multiple Shards (and you'd use whatever communication channel you have to make sure the other human is getting back the correct one). Or, we could just send them all as you suggest...

Reply:
all sounds like good ideas. especially if the 'petname' never needs to go over the wire.

Reply:
yeah I don't think the petname would need to go over the wire (which I guess might be sensitive, like if you chose `"that jerk I don't like from class"` as your petname haha).